libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs

```diff
@@ -231,35 +232,37 @@ private void InitializeModel()
 
             if (_snapshotFile == null)
             {
-                throw new ArgumentNullException($"Missing `ShapshotFile` information.");
+                throw new ArgumentNullException($"Missing `SnapshotFile` information.");
             }
 
-            if (orchestrator == null && _resolver == null)
+            if (_resolver != null)
             {
-                var fullModelPath = Path.GetFullPath(PathUtils.NormalizePath(_modelFolder));
+                return;
```

Comment:
(Let me preface, I think this is a fringe scenario, feel free to ignore).
I wonder if in this case we should verify consistency with passed in snapshot/model?